src/com/sahokia/kos/concerttickets/model/Ticket.java

```diff
@@ -73,52 +78,102 @@ public Ticket(String concertHall, String eventCode, boolean isPromo,
         }
     }
 
-    public String getId() {
-        return id;
-    }
-
+    @Override
     public void setId(String id) {
         if (id.length() <= 4) {
-            this.id = id;
+            super.setId(id);
```

Comment:
what is the difference between             this.setId(id); and current variant?

Reply:
Id was set as private in superclass before. Now I changed it to protected and return "this.id = id" variant